diff --git a/testing/taskcluster/scripts/phone-builder/build-phone.sh b/testing/taskcluster/scripts/phone-builder/build-phone.sh
--- a/testing/taskcluster/scripts/phone-builder/build-phone.sh
+++ b/testing/taskcluster/scripts/phone-builder/build-phone.sh
@@ -1,17 +1,14 @@
 #! /bin/bash -vex
 
 . pre-build.sh
 
-if [ $TARGET == "aries" -o $TARGET == "shinano" ]; then
-  # caching objects might be dangerous for some devices (aka aries)
-  rm -rf $WORKSPACE/B2G/objdir*
-  rm -rf $WORKSPACE/B2G/out
-fi
+rm -rf $WORKSPACE/B2G/objdir*
+rm -rf $WORKSPACE/B2G/out
 
 MOZHARNESS_CONFIG=${MOZHARNESS_CONFIG:=b2g/taskcluster-phone.py}
 
 rm -rf $WORKSPACE/B2G/upload/
 
 $WORKSPACE/gecko/testing/mozharness/scripts/b2g_build.py \
   --config $MOZHARNESS_CONFIG \
   "$debug_flag" \
diff --git a/testing/taskcluster/scripts/phone-builder/post-build.sh b/testing/taskcluster/scripts/phone-builder/post-build.sh
--- a/testing/taskcluster/scripts/phone-builder/post-build.sh
+++ b/testing/taskcluster/scripts/phone-builder/post-build.sh
@@ -1,17 +1,13 @@
 #! /bin/bash -vex
 
 # Don't cache backups
 rm -rf $WORKSPACE/B2G/backup-*
 
-if [ -f balrog_credentials ]; then
-  rm -f balrog_credentials
-fi
-
 mkdir -p $HOME/artifacts
 mkdir -p $HOME/artifacts-public
 
 DEVICE=${TARGET%%-*}
 
 mv $WORKSPACE/B2G/upload/sources.xml $HOME/artifacts/sources.xml
 mv $WORKSPACE/B2G/upload/b2g-*.android-arm.tar.gz $HOME/artifacts/b2g-android-arm.tar.gz
 mv $WORKSPACE/B2G/upload/${TARGET}.zip $HOME/artifacts/${TARGET}.zip
diff --git a/testing/taskcluster/scripts/phone-builder/pre-build.sh b/testing/taskcluster/scripts/phone-builder/pre-build.sh
--- a/testing/taskcluster/scripts/phone-builder/pre-build.sh
+++ b/testing/taskcluster/scripts/phone-builder/pre-build.sh
@@ -12,28 +12,21 @@ test $GECKO_HEAD_REPOSITORY # Should be 
 test $GECKO_BASE_REPOSITORY # Should be an hg repository url to clone from
 test $GECKO_HEAD_REV # Should be an hg revision to pull down
 test $TARGET
 test $VARIANT
 
 . ../builder/setup-ccache.sh
 
 # Figure out where the remote manifest is so we can use caches for it.
-MANIFEST=${MANIFEST:-`repository-url.py $GECKO_HEAD_REPOSITORY $GECKO_HEAD_REV b2g/config/$TARGET/sources.xml`}
+MANIFEST=${http://b2g-community-builds.github.io/raw-file/manifests/${TARGET}.xml}
 tc-vcs repo-checkout $WORKSPACE/B2G https://git.mozilla.org/b2g/B2G.git $MANIFEST
 
 # Ensure symlink has been created to gecko...
 rm -f $WORKSPACE/B2G/gecko
 ln -s $WORKSPACE/gecko $WORKSPACE/B2G/gecko
 
 debug_flag=""
 if [ 0$B2G_DEBUG -ne 0 ]; then
   debug_flag='--debug'
 fi
 
-backup_file=$(aws --output=text s3 ls s3://b2g-phone-backups/$TARGET/ | tail -1 | awk '{print $NF}')
-
-if echo $backup_file | grep '\.tar\.bz2'; then
-    aws s3 cp s3://b2g-phone-backups/$TARGET/$backup_file .
-    tar -xjf $backup_file -C $WORKSPACE/B2G
-    rm -f $backup_file
-fi
-
+curl -L http://b2g-community-builds.github.io/blobs-zips/$TARGET.bz2 -o $WORKSPACE/$TARGET.bz2
